test/tools/check_compilation.py

```diff
@@ -101,7 +102,7 @@ def getLogErrors(log):
 			elif terminatedError.search(line):
 				terminated = True
 			else:
-				print line,
+				sys.stdout.write (line)
```

Comment:
Not a python expert but wouldn't we lose the space between printed lines if we directly use sys.stdout.write?

Reply:
The original Python 2 statement `print line,` with a trailing comma, would have skipped the newline. `sys.stdout.write` gives the same behavior, and works with both Python 2 and 3.

Reply:
Newline is fine and it works with both 2 and 3. The question is about space.

I was trying the python program below and with python 2 it gave the following output. Where the print with comma added space but sys.stdout.write does not add a space.
one two three
onetwothree
```
import sys
mylist = ["one", "two", "three"]
for l in mylist:
    print l,
print ""
for l in mylist:
    sys.stdout.write(l)
```

Reply:
yes. this may be a problem and print (line, endl=" ") is the correct code for python 3.x. 
Looks like I need to add additional sys.stdout.write(" ")  line. would that be fine?